### PR TITLE
Basic info

### PR DESCRIPTION
**Pull request by TheSlimvReal**
see issue: #510 

- Moved Basic-Info to separate component
- Created abstract class which handles most of the logic for all form subcomponents
- Changed health and dropout form to use reactive forms instead of two-way-bindings

### Visible/Frontend Changes
- Each form subcomponent has its own `change`, `edit` and `cancel` button
- `Delete` button moved to end of page

### Architectural/Backend Changes
- [x] `Basic-Info`, `Health` and `Dropout` are now separate but quite similar subcomponents

### CODE REVIEW

File: src/app/child-dev-project/children/child-details/form-components/form-subcomponent.ts [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+  public constructor(
+    private entityMapperService: EntityMapperService,
+    private fb: FormBuilder,
+    private alertService: AlertService
+  ) {
+    this.initForm(this.getFormConfig());
+  }
+
+  protected abstract getFormConfig(): {
+    controlsConfig: { [key: string]: any };
```
Comment (sleidig) [inline src/app/child-dev-project/children/child-details/form-components/form-subcomponent.ts:22]: can we type these more specifically than `any`?

Comment (TheSlimvReal) [reply src/app/child-dev-project/children/child-details/form-components/form-subcomponent.ts:22]: Its the type definition for the arguments of the `fb.group(...)`. I dont think we can make it any more specific

**Comment by TheSlimvReal** [comment]
replaced with #526

**Comment by TheSlimvReal** [comment]
replaced by #526

**Comment by TheSlimvReal** [comment]
replaced by #526

**Pull request closed without merging**